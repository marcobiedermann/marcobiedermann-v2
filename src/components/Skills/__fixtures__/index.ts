import skill from '../../Skill/__fixtures__';

const skills = [skill];

export default skills;
