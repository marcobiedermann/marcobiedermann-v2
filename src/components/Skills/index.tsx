import classNames from 'classnames';
import React, { FC } from 'react';
import Skill, { SkillProps } from '../Skill';
import styles from './style.module.css';

export interface SkillsProps {
  className?: string;
  skills: SkillProps[];
}

export const Skills: FC<SkillsProps> = (props) => {
  const { className, skills } = props;

  return (
    <ul className={classNames(className, styles.skills)}>
      {skills.map((skill) => (
        <li key={skill.id}>
          <Skill {...skill} />
        </li>
      ))}
    </ul>
  );
};

export default Skills;
