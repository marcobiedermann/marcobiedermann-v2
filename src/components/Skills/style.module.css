.skills {
  display: grid;
  grid-gap: 1.5em;
  grid-template-columns: repeat(auto-fill, minmax(22em, 1fr));
  list-style: none;
  margin-bottom: 1.5em;
  margin-top: 0.5em;
  padding: 0;
}
