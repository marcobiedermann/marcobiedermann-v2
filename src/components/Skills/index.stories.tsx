/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Skills from '.';
import skills from './__fixtures__';

storiesOf('Skills', module).add('default', () => <Skills skills={skills} />);
