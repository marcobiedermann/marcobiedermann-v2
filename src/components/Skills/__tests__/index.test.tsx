/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Skills from '..';
import skills from '../__fixtures__';

describe('Skills component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Skills skills={skills} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
