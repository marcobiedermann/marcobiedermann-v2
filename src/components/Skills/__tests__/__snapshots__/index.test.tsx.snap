// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills component should render correctly 1`] = `
<DocumentFragment>
  <ul
    class="skills"
  >
    <li>
      <div>
        <h3>
          JavaScript
        </h3>
        <div />
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
